x-pack/legacy/plugins/siem/public/pages/404.tsx

```diff
@@ -5,17 +5,17 @@
  */
 
 import React from 'react';
-import { pure } from 'recompose';
 import { FormattedMessage } from '@kbn/i18n/react';
 
 import { WrapperPage } from '../components/wrapper_page';
 
-export const NotFoundPage = pure(() => (
+export const NotFoundPage = React.memo(() => (
```

Comment:
We will get an error form react-router here, it only allows pure function.

Reply:
Here's the issue in question: https://github.com/ReactTraining/react-router/issues/6471

It's fixed in React Router v4.4, but not sure what that upgrade path looks like for us... 😬

Reply:
For now, it appears this line fixes in our version of RR: https://github.com/elastic/kibana/pull/50806/files#diff-5a58397aa411ae017e0e1b99e0b379faL23